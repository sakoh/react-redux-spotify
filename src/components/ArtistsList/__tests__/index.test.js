import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ArtistList from '..'

configure({ adapter: new Adapter() })

describe('ArtistList', () => {
  it('should render correctly', () => {
    const artists = [
      {
        name: 'Iron Maiden',
        image: 'http://via.placeholder.com/350x150',
        popularity: 10,
      },
      {
        name: 'Metallica',
        image: 'http://via.placeholder.com/350x150',
        popularity: 10,
      },
      {
        name: 'Eminenm',
        image: 'http://via.placeholder.com/350x150',
        popularity: 10,
      },
    ]
    const wrapper = shallow(
      <ArtistList title='Great Artists' artists={artists} />  
    )

    expect(wrapper).toMatchSnapshot()
  })
})
