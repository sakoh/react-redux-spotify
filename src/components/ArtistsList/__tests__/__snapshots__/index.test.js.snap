// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistsList
    artists={
        Array [
            Object {
              "image": "http://via.placeholder.com/350x150",
              "name": "Iron Maiden",
              "popularity": 10,
            },
            Object {
              "image": "http://via.placeholder.com/350x150",
              "name": "Metallica",
              "popularity": 10,
            },
            Object {
              "image": "http://via.placeholder.com/350x150",
              "name": "Eminenm",
              "popularity": 10,
            },
          ]
    }
    title="Great Artists"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Great Artists
</h1>,
        <Unknown>
          <Unknown
                    image="http://via.placeholder.com/350x150"
                    name="Iron Maiden"
                    popularity={10}
          />
          <Unknown
                    image="http://via.placeholder.com/350x150"
                    name="Metallica"
                    popularity={10}
          />
          <Unknown
                    image="http://via.placeholder.com/350x150"
                    name="Eminenm"
                    popularity={10}
          />
</Unknown>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Great Artists",
        },
        "ref": null,
        "rendered": "Great Artists",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Unknown
              image="http://via.placeholder.com/350x150"
              name="Iron Maiden"
              popularity={10}
/>,
            <Unknown
              image="http://via.placeholder.com/350x150"
              name="Metallica"
              popularity={10}
/>,
            <Unknown
              image="http://via.placeholder.com/350x150"
              name="Eminenm"
              popularity={10}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "image": "http://via.placeholder.com/350x150",
              "name": "Iron Maiden",
              "popularity": 10,
            },
            "ref": null,
            "rendered": null,
            "type": undefined,
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "image": "http://via.placeholder.com/350x150",
              "name": "Metallica",
              "popularity": 10,
            },
            "ref": null,
            "rendered": null,
            "type": undefined,
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "image": "http://via.placeholder.com/350x150",
              "name": "Eminenm",
              "popularity": 10,
            },
            "ref": null,
            "rendered": null,
            "type": undefined,
          },
        ],
        "type": undefined,
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Great Artists
</h1>,
          <Unknown>
            <Unknown
                        image="http://via.placeholder.com/350x150"
                        name="Iron Maiden"
                        popularity={10}
            />
            <Unknown
                        image="http://via.placeholder.com/350x150"
                        name="Metallica"
                        popularity={10}
            />
            <Unknown
                        image="http://via.placeholder.com/350x150"
                        name="Eminenm"
                        popularity={10}
            />
</Unknown>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Great Artists",
          },
          "ref": null,
          "rendered": "Great Artists",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Unknown
                image="http://via.placeholder.com/350x150"
                name="Iron Maiden"
                popularity={10}
/>,
              <Unknown
                image="http://via.placeholder.com/350x150"
                name="Metallica"
                popularity={10}
/>,
              <Unknown
                image="http://via.placeholder.com/350x150"
                name="Eminenm"
                popularity={10}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "image": "http://via.placeholder.com/350x150",
                "name": "Iron Maiden",
                "popularity": 10,
              },
              "ref": null,
              "rendered": null,
              "type": undefined,
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "image": "http://via.placeholder.com/350x150",
                "name": "Metallica",
                "popularity": 10,
              },
              "ref": null,
              "rendered": null,
              "type": undefined,
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "image": "http://via.placeholder.com/350x150",
                "name": "Eminenm",
                "popularity": 10,
              },
              "ref": null,
              "rendered": null,
              "type": undefined,
            },
          ],
          "type": undefined,
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
